### ISSUE TITLE
PulseAudio is available for non-linux systems

### ISSUE THREAD

**Issue by falkTX** [thestk/rtaudio#116]
The current code makes it sound like pulseaudio is a linux specific API.
But I was able to build rtaudio with pulseaudio enabled for freebsd and gnu-hurd.

Would be nice to rename LINUX_PULSE stuff to UNIX_PULSE.

**Comment by radarsat1** [comment]
While I don't disagree, I'm not sure I see the advantage of possibly breaking configuration / build system code for other people's projects for such cosmetic reasons.  Let's put this on the pile along with C++11 of "things we'll break when we decide to break everything"

**Comment by falkTX** [comment]
Makes sense..
I personally renamed the macro to make it less confusing, but it will be a breaking change for sure.

**Comment by tay10r** [comment]
If we're talking about renaming the macros, I would go so far as to say drop the operating system names entirely. I would also like to suggest removing the underscores in the beginning of the identifiers and making the macros more descriptive. Here's what I would suggest for the header and build systems.

```c++
#define RTAUDIO_WITH_JACK
#define RTAUDIO_WITH_ALSA
#define RTAUDIO_WITH_CORE
#define RTAUDIO_WITH_PULSE
#define RTAUDIO_WITH_WASAPI
#define RTAUDIO_WITH_ASIO
#define RTAUDIO_WITH_DS // Or RTAUDIO_WITH_DIRECTSOUND
```

Or something close to that.

**Comment by Parakleta** [comment]
FreeBSD needs to use /usr/local/lib or /usr/local/include to access the pulseaudio libraries.

**Comment by ntonnaett** [comment]
With #313 API defines shouldn't be necessary anymore. So we would only have to change RtAudio's build systems.

**Comment by falkTX** [comment]
I saw that PR changes just now, doesn't seem complete.

**Comment by ntonnaett** [comment]
What is missing?

**Comment by falkTX** [comment]
I left a few comments in the code, hopefully github is able to show them this time...

**Comment by ntonnaett** [comment]
I've actually meant #308 but your point there is true for both.

**Comment by radarsat1** [comment]
Closing as solved with #308 and #313.
